Tools/CISupport/ews-build/steps.py

```diff
@@ -4769,6 +4770,61 @@ def start(self):
         return None
 
 
+class ValidateRemote(shell.ShellCommand):
+    name = 'validate-remote'
+    haltOnFailure = False
+    flunkOnFailure = True
+
+    def __init__(self, **kwargs):
+        self.summary = ''
+        super(ValidateRemote, self).__init__(logEnviron=False, **kwargs)
+
+    def start(self, BufferLogObserverClass=logobserver.BufferLogObserver):
+        base_ref = self.getProperty('github.base.ref', f'origin/{DEFAULT_BRANCH}')
+        remote = self.getProperty('remote', 'origin')
+
+        self.command = [
+            'git', 'merge-base', '--is-ancestor',
+            f'remotes/{remote}/{base_ref}',
+            f'remotes/origin/{base_ref}',
+        ]
+
+        return super(ValidateRemote, self).start()
+
+    def getResultSummary(self):
+        if self.results in (FAILURE, SUCCESS):
+            return {'step': self.summary}
+        return super(ValidateRemote, self).getResultSummary()
+
+    def evaluateCommand(self, cmd):
+        base_ref = self.getProperty('github.base.ref', f'origin/{DEFAULT_BRANCH}')
+        rc = super(ValidateRemote, self).evaluateCommand(cmd)
+
+        if rc == SUCCESS:
+            self.summary = f"Cannot land on '{base_ref}', it is owned by 'origin'"
```

Comment:
This error message seems very confusing. What does it mean to be owned by 'origin'. Whom should it be owned by? Should be more explanative.